src/main/java/com/uber/cadence/worker/Worker.java

```diff
@@ -49,95 +53,44 @@
   private final SyncWorkflowWorker workflowWorker;
   private final SyncActivityWorker activityWorker;
   private final AtomicBoolean started = new AtomicBoolean();
+  private final AtomicBoolean closed = new AtomicBoolean();
 
   /**
-   * Creates worker that connects to the local instance of the Cadence Service that listens on a
-   * default port (7933).
-   *
-   * @param domain domain that worker uses to poll.
-   * @param taskList task list name worker uses to poll. It uses this name for both decision and
-   *     activity task list polls.
-   */
-  public Worker(String domain, String taskList) {
-    this(domain, taskList, null);
-  }
-
-  /**
-   * Creates worker that connects to the local instance of the Cadence Service that listens on a
-   * default port (7933).
+   * Creates worker that connects to an instance of the Cadence Service.
    *
+   * @param service client to the Cadence Service endpoint.
    * @param domain domain that worker uses to poll.
    * @param taskList task list name worker uses to poll. It uses this name for both decision and
    *     activity task list polls.
    * @param options Options (like {@link DataConverter} override) for configuring worker.
    */
-  public Worker(String domain, String taskList, WorkerOptions options) {
-    this(new WorkflowServiceTChannel(), domain, taskList, options);
-  }
+  private Worker(IWorkflowService service, String domain, String taskList, WorkerOptions options) {
+    Preconditions.checkNotNull(service, "service should not be null");
```

Comment:
The Java 8 introduced Objects.requireNotNull for this purpose:

https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html